pkg/util/containers/v2/metrics/containerd/collector.go

```diff
@@ -71,13 +71,9 @@ func (c *containerdCollector) ID() string {
 }
 
 // GetContainerStats returns stats by container ID.
-func (c *containerdCollector) GetContainerStats(containerID string, cacheValidity time.Duration) (*provider.ContainerStats, error) {
-	namespace, err := c.containerNamespace(containerID)
-	if err != nil {
-		return nil, err
-	}
-	c.client.SetCurrentNamespace(namespace)
-
+func (c *containerdCollector) GetContainerStats(containerNS, containerID string, cacheValidity time.Duration) (*provider.ContainerStats, error) {
+	// TODO: Relying on `SetCurrentNamespace` is not correct as the collector is supposed to allow for concurrent calls
```

Comment:
Mind clarifying this @vboulineau? Do you plan to address in this PR?

Reply:
No, it's unrelated to this PR, the `containerd` collector could have a bug if used concurrently (which is not the case currently), just added the comment to point the exact place in the code this could cause issues.